build_platform.py

```diff
@@ -428,6 +460,89 @@ def test_examples_in_folder(platform, folderpath):
             success = 1
 
 
+def install_boards_local_txt(core_fqbn, boards_local_txt):
+    """Copies boards.local.txt to the appropriate folder for the given core_fqbn.
+    :param str core_fqbn: The first two segments of fully qualified board
+      name, vendor:architecture (e.g., "adafruit:samd").
+    :param str boards_local_txt: The path to the boards.local.txt file.
+    """
+    try:
+        local_app_data_dir = os.environ.get('HOME', '')
+        data_dir = None
```

Comment:
place `local_app_data_dir` and `data_dir` above the try, it's referenced lower in the code too. If it fails, you dont have a reference to these when you need data_dir again on L479